core/build.gradle

```diff
@@ -1,11 +1,15 @@
 apply plugin: "java"
 
-sourceCompatibility = 1.6
+sourceCompatibility = 1.7
 [compileJava, compileTestJava]*.options*.encoding = 'UTF-8'
 
-sourceSets.main.java.srcDirs = [ "src/" ]
-
+sourceSets.main.java.srcDirs = ["src/main/java"]
+sourceSets.test.java.srcDirs = ["src/test/java"]
 
 eclipse.project {
     name = appName + "-core"
 }
+
+dependencies {
+    testCompile 'junit:junit:4.12'
+}
```

Comment:
The dependency should be in the main build.gradle. Where we have all dependencies for the core/desktop project.